### PR TITLE
Feat aanpassen rooster

### PR DESCRIPTION
**Pull request by MaxdeVriesHSL**
## Description
Feature that allows admins to view a table of all planned shifts. These are sorted by date and time, and display the current staffed users.

Shifts can be expanded, displaying more details and allowing the admin to change the properties of the shift, such as:
- start time
- end time
- required amount of volunteers for the generate schedule to assign
- delete shift

In addition to this, the admin can manually change staffing assigned to a shift by doing the following:
- remove staffing
- select a volunteer/employee to add as staffing

## What changed?
Added Role.EMPLOYEE in role.ts and made a duo of mock employees in seed.ts

## How to test:
1. Navigate to the branch feat_AanpassenRooster and pull.
2. Ensure you can log in as admin by looking at your env file.
3. Run it with "npm run seed" followed up by "npm run dev".
4. Go to the admin panel and press "Handmatige aanpassingen".
5. You will see a blank table, these are shifts where no one has been assigned as staffing.
6. Go back to the admin panel and generate roster for the current month.
7. Back again to the shift overview and now you'll see all staffed users.
8. Click on a row and it will expand.
9. Now you can do the following:
- Press the button on the top left to minimize the shift
- Press the button the top right to delete the shift
- Change the Starttijd field and press on the button next to it
- Change the Eindtijd field and press on the button next to it
- Click on the + / - buttons under Benodigde vrijwilligers and press the button next to it
- Press the trash can button next to an assigned staffing to remove it
- Select a shift type at the selector at the bottom of the shift
- Select a volunteer or employee, based on the selected shift type (if a user is staffed already, you won't see them in the dropdown so you can't add them twice, if you remove a staffing you'll see you can select that user again in the dropdown.)
- Press the button below the select field to add the staffing to the shift
10. Remember your changes and refresh the page.
11. If you expand the same shift again you'll see everything is still there 👍

## Some screenshots if you're lazy:
Overview shifts:
image

Expanded shift:
image

Changed time:
image

Changed staffing required:
image

Feedback of properties changed:
image

Adding a volunteer:
image

Volunteer added:
image

New shift overview with updated properties:
image

### CODE REVIEW

**Comment by MaxdeVriesHSL** [comment]
Zet 'm ff op closed, was een poging om het klaar te zetten voor sprint review maar is niet af.

**Comment by MaxdeVriesHSL** [comment]
We back boys

**Comment by MaxdeVriesHSL** [comment]
Hold up need to solve merge conflicts with dev 💀

File: src/pages/admin/shiften.tsx [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+
+    setDateValueStart(start)
+    setDateValueEnd(end)
+  }
+
+  const getFormattedTimeShift = (shiftTime: Date): string => {
+    const dateTime = new Date(shiftTime)
+
+    const year = String(dateTime.getFullYear())
+    const month = String(dateTime.getMonth() + 1).padStart(2, "0")
```
Comment (wardalbam) [inline src/pages/admin/shiften.tsx:135]: What doet deze code?!

**Review by wardalbam — APPROVED** [review]
Lekker beziggg.
Alles ziet er goed uit!!

File: src/pages/admin/shiften.tsx (hunk 1 above)
Comment (MaxdeVriesHSL) [reply src/pages/admin/shiften.tsx:135]: De functie schrijft een Date om naar een String, omdat andere methods dit nodig hebben. Dit moet volgens het format '2023-06-21T14:58'.
Stel je voor dat je de datum 1 januari pakt met de tijd 5 over 9 's ochtends zou dit zonder de .padStart er als volgt uitzien: 1/1 9/5
Je wil hier natuurlijk 01/01 09:05 krijgen
In het geval van padStart(2,"0") voegt deze method een 0 toe aan het begin van de string tot deze een lengte van 2 bereikt
"9".padStart(2,"0") -> 09, "5".padStart(3,"0") -> 0005

**Pull request merged**